lib/modules/dosomething/dosomething_api/includes/ApiCache.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+class ApiCache {
+
+  /**
+   * Clear a specified item from cache.
+   *
+   * @param  mixed  $id
+   * @param  mixed  $endpoint
+   * @param  mixed  $parameters
+   * @return bool
+   */
+  function clear($id = NULL, $endpoint = NULL, $parameters = NULL) {
+    if (!$id) {
+      if ($endpoint && $parameters) {
+        $id = $this->generate_id($endpoint, $parameters);
+      }
+      else {
+        return FALSE;
+      }
+    }
+
+    cache_clear_all($id, 'cache_dosomething_api');
+
+    return TRUE;
+  }
+
+
+  /**
+   * Get data from cache based on id from endpoint and URL parameters.
+   *
+   * @param  string  $endpoint  Type of resource based on endpoint.
+   * @param  array   $parameters  The URL parameters passed that define the request.
+   * @return mixed
+   */
+  public function get($endpoint, $parameters) {
+    $id = $this->generate_id($endpoint, $parameters);
+
+    if (!dosomething_helpers_convert_string_to_boolean($parameters['cache'])) {
+      $this->clear($id);
+
+      return FALSE;
+    }
+
+    $cache = cache_get($id, 'cache_dosomething_api');
+
+    if ($cache && $cache->expire < REQUEST_TIME) {
+      $this->clear($id);
+
+      return FALSE;
+    }
+
+    return $cache;
+  }
+
+
+  /**
+   * Set data in cache for 24 hours with id based on endpoint and URL parameters.
+   *
+   * @param  string  $endpoint
+   * @param  array   $parameters
+   * @param  mixed   $data
+   * @return bool
+   */
+  public function set($endpoint, $parameters, $data) {
+    $id = $this->generate_id($endpoint, $parameters);
+
+    if (!dosomething_helpers_convert_string_to_boolean($parameters['cache'])) {
+      return FALSE;
+    }
+
+    cache_set($id, $data, 'cache_dosomething_api', REQUEST_TIME + (60 * 60 * 24));
```

Comment:
I wouldn't be opposed to moving this `(60 * 60 * 24)` up as a class property? As a property it could make it easier to spot that the cache time is 24 hours.